### PR TITLE
ADAL to MSAL - Advanced C#

### PR DESCRIPTION
**Pull request by cristobal-buenrostro**
Code sample updates:

use MSAL instead of ADAL for getting AAD token.
upgrade to .Net Core 6.

### CODE REVIEW

File: js/samples/advanced-csharp/Pages/Index.cshtml [hunk 1]
```diff
@@ -112,7 +112,22 @@
...
             const subdomain = await getImmersiveReaderSubdomainAsync();
 
-            ImmersiveReader.launchAsync(token, subdomain, data);
+            // Learn more about options https://docs.microsoft.com/azure/cognitive-services/immersive-reader/reference#options
+            const options = {
+                cognitiveAppId: 'AdvancedSampleWebApp',
+                onExit: exitCallback
+            };
+
+            try {
```
Comment (rwallerms) [inline js/samples/advanced-csharp/Pages/Index.cshtml:121]: try/catch? Doesn't this need Promise handling?

File: js/samples/advanced-csharp/Pages/Math.cshtml [hunk 2]
```diff
@@ -181,7 +181,22 @@
...
             const subdomain = await getImmersiveReaderSubdomainAsync();
 
-            ImmersiveReader.launchAsync(token, subdomain, data);
+            // Learn more about options https://docs.microsoft.com/azure/cognitive-services/immersive-reader/reference#options
+            const options = {
+                cognitiveAppId: 'AdvancedSampleWebApp',
+                onExit: exitCallback
+            };
+
+            try {
```
Comment (rwallerms) [inline js/samples/advanced-csharp/Pages/Math.cshtml:190]: try/catch?

File: js/samples/advanced-csharp/Pages/Document.cshtml [hunk 3]
```diff
@@ -75,7 +75,22 @@
             const token = await getImmersiveReaderTokenAsync(Canary);
             const subdomain = await getImmersiveReaderSubdomainAsync();
 
-            ImmersiveReader.launchAsync(token, subdomain, data);
+            // Learn more about options https://docs.microsoft.com/azure/cognitive-services/immersive-reader/reference#options
+            const options = {
+                cognitiveAppId: 'AdvancedSampleWebApp',
```
Comment (rwallerms) [inline js/samples/advanced-csharp/Pages/Document.cshtml:80]: cognitiveAppId is not an option we expose in the Public SDK, should remove this

File: js/samples/advanced-csharp/Pages/MultiLang.cshtml [hunk 4]
```diff
@@ -13,7 +13,7 @@
 
     <script type='text/javascript' src='https://code.jquery.com/jquery-3.3.1.min.js'></script>
     <script type='text/javascript' src='https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js'></script>
-    <script type='text/javascript' src='https://ircdname.azureedge.net/immersivereadersdk/immersive-reader-sdk.1.2.0.js'></script>
+<script type='text/javascript' src='https://ircdname.azureedge.net/immersivereadersdk/immersive-reader-sdk.1.2.0.js'></script>
```
Comment (rwallerms) [inline js/samples/advanced-csharp/Pages/MultiLang.cshtml:16]: indent

File: js/samples/advanced-csharp/wwwroot/css/styles.css [hunk 5]
```diff
@@ -25,9 +25,9 @@ nav {
     text-align: center;
 }
 
-nav a {
-    padding: 3px;
-}
+    nav a {
```
Comment (rwallerms) [inline js/samples/advanced-csharp/wwwroot/css/styles.css:28]: revert tabbing here?

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/advanced-csharp/Controllers/ApiController.cs [hunk 6]
```diff
@@ -55,36 +55,55 @@ public string GetSubdomain()
...
         }
 
-        protected async Task<string> GetTokenAsync()
+        /// <summary>
+        /// Get an Azure AD authentication token
+        /// </summary>
+        public async Task<string> GetTokenAsync()
         {
             string authority = $"https://login.windows.net/{TenantId}";
             const string resource = "https://cognitiveservices.azure.com/";
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Controllers/ApiController.cs:93]: I will revert in previous PR

File: js/samples/advanced-csharp/AdvancedSampleWebApp.csproj [hunk 7]
```diff
@@ -1,29 +1,15 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/AdvancedSampleWebApp.csproj]: removed

File: js/samples/advanced-csharp/AdvancedSampleWebApp.csproj [hunk 8]
```diff
@@ -1,29 +1,15 @@
...
+    <ImplicitUsings>enable</ImplicitUsings>
+    <UserSecretsId>3786c12f-3f68-4980-95ce-8fbe362c91d6</UserSecretsId>
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/AdvancedSampleWebApp.csproj]: removed

File: js/samples/advanced-csharp/Controllers/ApiController.cs [hunk 9]
```diff
@@ -15,6 +13,8 @@ public class ApiController : ControllerBase
         private readonly string ClientSecret; // Azure AD Application Service Principal password
         private readonly string Subdomain;    // Immersive Reader resource subdomain (resource 'Name' if the resource was created in the Azure portal, or 'CustomSubDomain' option if the resource was created with Azure CLI Powershell. Check the Azure portal for the subdomain on the Endpoint in the resource Overview page, for example, 'https://[SUBDOMAIN].cognitiveservices.azure.com/')
 
+        IConfidentialClientApplication app;
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Controllers/ApiController.cs]: changed to property

File: js/samples/advanced-csharp/Controllers/ApiController.cs [hunk 10]
```diff
@@ -15,6 +13,8 @@ public class ApiController : ControllerBase
         private readonly string ClientSecret; // Azure AD Application Service Principal password
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Controllers/ApiController.cs:13]: done

File: js/samples/advanced-csharp/Pages/Document.cshtml (hunk 3 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Pages/Document.cshtml]: removed for all html pages

File: js/samples/advanced-csharp/Pages/MultiLang.cshtml (hunk 4 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Pages/MultiLang.cshtml]: done

File: js/samples/advanced-csharp/wwwroot/css/styles.css (hunk 5 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/wwwroot/css/styles.css]: done

File: js/samples/advanced-csharp/Pages/Index.cshtml (hunk 1 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Pages/Index.cshtml:120]: changed to promise handling, for the rest of pages too

File: js/samples/advanced-csharp/Pages/Math.cshtml (hunk 2 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Pages/Math.cshtml:189]: changed for all pages

File: js/samples/advanced-csharp/Controllers/ApiController.cs [hunk 11]
```diff
@@ -13,7 +13,7 @@ public class ApiController : ControllerBase
...
         private readonly string Subdomain;    // Immersive Reader resource subdomain (resource 'Name' if the resource was created in the Azure portal, or 'CustomSubDomain' option if the resource was created with Azure CLI Powershell. Check the Azure portal for the subdomain on the Endpoint in the resource Overview page, for example, 'https://[SUBDOMAIN].cognitiveservices.azure.com/')
 
-        IConfidentialClientApplication app;
+        private IConfidentialClientApplication app { get; set; }
```
Comment (rwallerms) [inline js/samples/advanced-csharp/Controllers/ApiController.cs:16]: add this init code to the Property

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/advanced-csharp/Controllers/ApiController.cs (hunk 11 above)
Comment (cristobal-buenrostro) [reply js/samples/advanced-csharp/Controllers/ApiController.cs]: done, similar to the other samples

**Review by rwallerms — APPROVED** [review]

**Pull request merged**